orchestrator/coingecko/coingecko.go

```diff
@@ -22,35 +23,117 @@ const (
 
 var zeroPrice = float64(0)
 
-type PriceFeed struct {
+type CoinGecko struct {
 	client *http.Client
 	config *Config
 
-	interval time.Duration
+	coinsSymbol map[ethcmn.Address]string // contract addr => token symbol
 
 	logger zerolog.Logger
 }
 
+// Config wraps config variable to get information at CoinGecko
 type Config struct {
 	BaseURL string
 }
 
-func urlJoin(baseURL string, segments ...string) string {
+// CoinInfo wraps the coin information receveid from an contract address.
+// https://api.coingecko.com/api/v3/coins/ethereum/contract/${CONTRACT_ADDR}
+type CoinInfo struct {
+	Symbol string `json:"symbol"`
+	Error  string `json:"error"`
+}
+
+type priceResponse map[string]struct {
+	USD float64 `json:"usd"`
+}
+
+// NewCoingeckoPriceFeed returns price puller for given symbol. The price will be pulled
+// from endpoint and divided by scaleFactor. Symbol name (if reported by endpoint) must match.
+func NewCoingeckoPriceFeed(logger zerolog.Logger, endpointConfig *Config) *CoinGecko {
```

Comment:
We're not using coingecko as a price feed anymore as of this PR right? Just an address -> symbol converter? If so we should update this function name and godoc

Reply:
agreed, removed some stuffs at https://github.com/umee-network/peggo/pull/231/commits/45a4637b07ea74807bbc785eff14660d194d8ad3